src/data/nat/pairing.lean

```diff
@@ -132,3 +132,15 @@ lemma Union_unpair_prod {α β} {s : ℕ → set α} {t : ℕ → set β} :
 by { rw [← Union_prod], convert surjective_unpair.Union_comp _, refl }
 
 end set
+
+section complete_lattice
+
+lemma supr_unpair {α} [complete_lattice α] (f : ℕ → ℕ → α) :
+  (⨆ n : ℕ, f n.unpair.1 n.unpair.2) = ⨆ i j : ℕ, f i j :=
+by rw [← (supr_prod : (⨆ i : ℕ × ℕ, f i.1 i.2) = _), ← nat.surjective_unpair.supr_comp]
+
+lemma infi_unpair {α} [complete_lattice α] (f : ℕ → ℕ → α) :
+  (⨅ n : ℕ, f n.unpair.1 n.unpair.2) = ⨅ i j : ℕ, f i j :=
+supr_unpair (show ℕ → ℕ → order_dual α, from f)
```

Comment:
In your original application did you want the `Union` version? If so, you should probably add that one below these (proved in terms of them).

Reply:
Can I not directly apply the supr version in my use case?

Reply:
Not with a `rw`. It will work with an apply